routes/queryargs.go

```diff
@@ -0,0 +1,189 @@
+// a []int value. It can return an error, the error's message contains %s
+// which has to be replaced by the argument's name before being displayed.
+func (d QueryArgIntSlice) Parse(val string) (interface{}, error) {
+	vals := strings.Split(val, ",")
+	res := make([]int64, 0, len(vals))
+	for _, v := range vals {
+		if i, err := strconv.ParseInt(v, 10, 64); err == nil {
+			res = append(res, i)
+		} else {
+			return nil, errors.New("argument \"%s\" must be a slice of int")
+		}
+	}
+	return res, nil
+}
+
+// Parse is the method of QueryArgUintSlice allowing to type the row value to
+// a []uint value. It can return an error, the error's message contains %s
+// which has to be replaced by the argument's name before being displayed.
+func (d QueryArgUintSlice) Parse(val string) (interface{}, error) {
+	vals := strings.Split(val, ",")
+	res := make([]uint64, 0, len(vals))
+	for _, v := range vals {
+		if i, err := strconv.ParseUint(v, 10, 64); err == nil {
+			res = append(res, i)
+		} else {
+			return nil, errors.New("argument \"%s\" must be a slice of uint")
+		}
+	}
+	return res, nil
+}
+
+// Parse is the method of QueryArgStringSlice allowing to type the row value to
+// a []string value. It can't return an error since it only calls strings.Split().
+func (d QueryArgStringSlice) Parse(val string) (interface{}, error) {
+	return strings.Split(val, ","), nil
+}
+
+// Decorate is the function called to apply the decorators to an endpoint. It returns
+// a function. This function produces a 400 error code with a json error message or
+// calls the next IntermediateHandler.
+// The goal of this function is to get the URL parameters to store them in
+// the Arguments.
+//
+// To use this decorator, create a var:
+// 		QueryArgProductsName = QueryArg{"productsName", QueryArgStringSlice{}}
+//
+// In this example, we need a slice of string called productsName that will be
+// our products name so we indicate to the decorator with QueryArgStringSlice{}.
+//
+// Then, register your endpoint as usual:
+//			Register(
+//				"/products/cost",
+//				getProductsCost,
+//				RequireMethod{"GET"},
+//				RequireContentType{"application/json"},
+//				db.WithTransaction{db.Db},
+//				users.WithAuthenticatedUser{},
+//				WithQueryArg{QueryArgProductsName},
+//			)
+//
+// See the last decorator WithQueryArg{QueryArgProductsName}, it will store
+// a slice of product name in the arguments.
+//
+// Thus, if the user calls
+//		/products/cost?productsName=EC2,ES,RDS
+// you will be able to get the productsName with the key productsName in the
+// arguments. The value will be ["EC2", "ES", "RDS"].
+func (d WithQueryArg) Decorate(h IntermediateHandler) IntermediateHandler {
+	return func(w http.ResponseWriter, r *http.Request, a Arguments) (status int, output interface{}) {
+		for _, arg := range d {
+			if rawVal := r.URL.Query().Get(arg.Name); rawVal != "" {
+				if val, err := arg.Type.Parse(rawVal); err == nil {
+					a[arg.Name] = val
```

Comment:
`arg.Name` is a `string`. There is a risk of collisions if you don’t use some other type. You can use an unexported type and provide a function to retrieve the value in the handler, otherwise using `arg` as a key works just fine.